src/PlaylistSave.hxx

```diff
@@ -31,32 +31,38 @@ playlist_print_song(BufferedOutputStream &os, const DetachedSong &song);
 void
 playlist_print_uri(BufferedOutputStream &os, const char *uri);
 
-/**
- * Saves a queue object into a stored playlist file.
- */
-void
-spl_save_queue(const char *name_utf8, const Queue &queue);
+struct PlaylistSaveMode {
+private:
+	bool create = false;
+	bool replace = false;
+	bool append = false;
 
-/**
- * Saves a playlist object into a stored playlist file.
- */
-void
-spl_save_playlist(const char *name_utf8, const playlist &playlist);
+	const char *argValidationError = "Expected one of 'create', 'append', 'replace'";
+
+public:
+	[[nodiscard]] static PlaylistSaveMode parseArg(const char* arg);
+
+	[[nodiscard]] inline bool is_valid() const noexcept {
+		return create || (append != replace);
+	}
+
+	[[nodiscard]] inline bool is_create_only() const noexcept {
+		return create && !(append || replace);
+	}
+
+	[[nodiscard]] inline bool is_existing_only() const noexcept {
```

Comment:
More redundant `inline` - `inline` is implied for method bodies inside `struct{}`!